core/bundle/changes/handlers.go

```diff
@@ -57,6 +58,17 @@ func (r *resolver) handleApplications() (map[string]string, error) {
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		computedBase, err := getBase(application.Base, defaultBase, computedSeries)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		// Ensure computedSeries and base match
```

Comment:
Unfortunately we need both the base and series

1) If a base is not provided, we fall back to the series and convert to a base
2) Both base and series need to be passed into `Add.*Params` otherwise it'll be a breaking change with GUIArgs

Reply:
Why can't we transform a base into a series when Args() or GUIArgs() are called? Push transformation of the base to series to the edge rather than keeping it in the norm. Skip error checking on the results of series.GetSeriesFromBase(), for it to be in a change params, then it has already been validated.

If series is provided, transform it into a base when the bundle change is created. Limit the amount of series/base decision points in the code.

I'm not convinced you cannot get into a situation where the base and series are not equivalent.

Reply:
That's fair. I have now done so. I haven't skipped error checking for GUIArgs, instead I'm panicing. Hopefully this will further dissuade it and encourage us to promptly remove `GetChanges` (using `GetChangesMapArgs` insead which can't panic)

If you're against this though, happy to just ignore errors, but I do think a panic makes sense here (`mustBaseToSeries`)